app/service/data_svc.py

```diff
@@ -131,3 +136,24 @@ def __init__(self, dao):
               'ON a.ability_id=b.id ' \
               'WHERE a.op_id = %s;' % op_id
         return await self.dao.raw_select(sql)
+
+def make_uuid(_filename):
+    uuid_string = _filename.split('/')[-1].split('.')[0]
+    try:
+        val = UUID(uuid_string, version=4)
+        return _filename
+    except ValueError:
+        _uuid = str(uuid4())
+        new_ability_file = _filename.split('/')[:-1]
+        new_ability_file.append('{}.yml'.format(_uuid))
+        new_ability_file = '/'.join(new_ability_file)
+        with open(_filename, 'r') as ability:
+            with open(new_ability_file, 'w') as new_ability:
+                for line in ability:
+                    if '- id:' in line:
```

Comment:
this doesn't feel right, seems hacky. we should try a way that selects the ID from the YML (like a key in JSON), likely through the 'yml' library imported at the top of this file: yaml